packages/storybook/src/clinical/organisms/questionnaire/questionnaire.test.tsx

```diff
@@ -0,0 +1,105 @@
+import Questionnaire from '@ltht-react/questionnaire'
+import { DetailViewType, QuestionnaireResponse } from '@ltht-react/types'
+import { render, screen } from '@testing-library/react'
+import {
+  BooleanFieldData,
+  ComposedExampleData as NestedQuestionnaireData,
+  NoAnswerData,
+} from './questionnaire.fixtures'
+
+describe('Questionnaire', () => {
+  it('should render a question', async () => {
+    render(<Questionnaire questionnaire={BooleanFieldData} />)
+
+    await screen.findByText('Do you have allergies?')
+    screen.getByText('Yes')
+  })
+
+  it('should show question groups', async () => {
+    render(<Questionnaire questionnaire={NestedQuestionnaireData} />)
+
+    await screen.findByText('Do you have allergies?')
+    screen.getByText('Yes')
+
+    screen.getByText('General questions')
+
+    screen.getByText('What is your gender?')
+    screen.getByText(/Male/)
+  })
+
+  describe('compact view', () => {
+    it("should not show a question if it doesn't have an answer", async () => {
+      const data: QuestionnaireResponse = {
+        ...BooleanFieldData,
+        item: [
+          {
+            linkId: '1',
+            text: null,
+            answer: [],
+            item: null,
+          },
+        ],
+      }
+
+      render(
+        <>
+          <span>Render check</span>
+          <Questionnaire questionnaire={data} viewType={DetailViewType.Compact} />
+        </>
+      )
+
+      await screen.findByText('Render check')
+      expect(screen.queryByText('Do you have allergies?')).toBeNull()
+    })
+
+    it('should not show a group header when the children do not have answers', async () => {

```

Comment:
Bit of a minor nitpicky point but these test names don't mention if they apply to Compact or Expanded view, and it might be worth describing that if behaviour differs at all

Reply:
They're nested inside a `describe` with the name "compact view". Happy to add the view type here too if that it improves readability